src/linter/block.go

```diff
@@ -1027,7 +1027,202 @@ func (b *blockWalker) handleIssetDimFetch(e *ir.ArrayDimFetchExpr) {
 	}
 }
 
+func nullSafetyRealParamForCheck(fn meta.FuncInfo, paramIndex int, haveVariadic bool) meta.FuncParam {
+	if haveVariadic && paramIndex >= len(fn.Params)-1 {
+		return fn.Params[len(fn.Params)-1]
+	}
+	return fn.Params[paramIndex]
+}
+
+func formatSlashesFuncName(fn meta.FuncInfo) string {
+	return strings.TrimPrefix(fn.Name, "\\")
+}
+
+func (b *blockWalker) checkNullSafetyCallArgsF(args []ir.Node, fn meta.FuncInfo) {
+	if fn.Params == nil || fn.Name == "" {
+		return
+	}
+	haveVariadic := fn.Flags&meta.FuncVariadic != 0
+
+	for i, arg := range args {
+		if arg == nil {
+			continue
+		}
+
+		// If there are more arguments than declared and function is not variadic, ignore extra arguments.
+		if i > len(fn.Params)-1 && !haveVariadic {
```

Comment:
do you want to change `!haveVariadic && i > len(fn.Params)-1` to be the same everywhere?

Reply:
Ok